tests/0143-exponential_backoff.c

```diff
@@ -0,0 +1,418 @@
+                                if (high > ((retry_max_ms*125)/100))
+                                        high = (retry_max_ms*125)/100;
+                                if (low > ((retry_max_ms*75)/100))
+                                        low = (retry_max_ms*75)/100;
+                                TEST_ASSERT((time_difference < high) && (time_difference > low),"Time difference is not respected!\n");
+                                retry_count++;
+                        }
+                        previous_request_ts = rd_kafka_mock_request_timestamp(requests[i]);
+                }
+        }
+        rd_kafka_topic_destroy(rkt);
+        rd_kafka_destroy(producer); 
+        rd_kafka_mock_clear_requests(mcluster);
+        SUB_TEST_PASS();
+}
+
+/** Fetch-FastLeaderQuery Test
+ * We fail the request with leader change or epoch to be precise so that it triggers FastLeaderQuery request.
+*/
+void test_Fetch_FastLeaderQuery(rd_kafka_mock_cluster_t *mcluster,const char *topic, rd_kafka_conf_t *conf){
+        rd_kafka_mock_request_t **requests = NULL;
+        size_t request_cnt = 0;
+        rd_bool_t fetched = rd_false;
+        rd_bool_t flag = rd_false;
+        rd_kafka_t *consumer;
+        SUB_TEST();
+        test_conf_set(conf, "topic.metadata.refresh.interval.ms","-1");
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+        test_conf_set(conf, "enable.auto.commit", "false");
+
+        consumer = test_create_consumer(topic, NULL, rd_kafka_conf_dup(conf), NULL);
+        
+        test_consumer_subscribe(consumer,topic);
+        rd_kafka_message_t *rkm = rd_kafka_consumer_poll(consumer,10*1000);
+        if (rkm)
+                rd_kafka_message_destroy(rkm);
+        rd_kafka_mock_clear_requests(mcluster);
+        
+        rd_kafka_mock_partition_set_leader(mcluster,topic,0,1);
+        rkm = rd_kafka_consumer_poll(consumer,10*1000);
+        if (rkm)
+                rd_kafka_message_destroy(rkm);
+        rd_sleep(3);
+        requests = rd_kafka_mock_get_requests(mcluster,&request_cnt);
+        for(size_t i = 0; i < request_cnt; i++){
+                TEST_SAY("Broker Id : %d API Key : %d Timestamp : %lld\n",rd_kafka_mock_request_id(requests[i]), rd_kafka_mock_request_api_key(requests[i]), rd_kafka_mock_request_timestamp(requests[i]));
+                if(rd_kafka_mock_request_api_key(requests[i]) == RD_KAFKAP_Fetch)
+                        fetched = rd_true;
+                else if(rd_kafka_mock_request_api_key(requests[i]) == RD_KAFKAP_Metadata && fetched){
+                        flag = rd_true;
+                        break;
+                }else{
+                        fetched = rd_false;
+                }
+        }
+        rd_kafka_destroy(consumer); 
+        rd_kafka_mock_clear_requests(mcluster);
+        TEST_ASSERT((flag==rd_true),"Metadata Request should have been made after fetch atleast once.\n");
```

Comment:
Don't call it flag, give it a more descriptive name.

also, for booleans, it is redundant to assert like `TEST_ASSERT(boolean == rd_true)`. Just do `TEST_ASSERT(boolean, ...)`